WordPress/Classes/ViewRelated/Menus/Views/MenuItemSourceCell.m

```diff
@@ -0,0 +1,165 @@
+                                              [stackView.trailingAnchor constraintLessThanOrEqualToAnchor:self.contentView.trailingAnchor],
+                                              [stackView.bottomAnchor constraintEqualToAnchor:self.contentView.bottomAnchor]
+                                              ]];
+    
+    _stackView = stackView;
+}
+
+- (void)initLabelsStackView
+{
+    UIStackView *labelsStackView = [[UIStackView alloc] init];
+    labelsStackView.translatesAutoresizingMaskIntoConstraints = NO;
+    labelsStackView.distribution = UIStackViewDistributionFill;
+    labelsStackView.alignment = UIStackViewAlignmentTop;
+    labelsStackView.axis = UILayoutConstraintAxisHorizontal;
+    labelsStackView.spacing = self.stackView.spacing;
+    _labelsStackView = labelsStackView;
+
+    NSAssert(_stackView != nil, @"stackView is nil");
+    [_stackView addArrangedSubview:labelsStackView];
+}
+
+- (void)initLabel
+{
+    UILabel *label = [[UILabel alloc] init];
+    label.translatesAutoresizingMaskIntoConstraints = NO;
+    label.font = [WPStyleGuide tableviewTextFont];
+    label.textColor = [WPStyleGuide greyDarken30];
+    label.backgroundColor = [UIColor whiteColor];
+    label.numberOfLines = 0;
+    label.lineBreakMode = NSLineBreakByTruncatingTail;
+    
+    [label setContentHuggingPriority:UILayoutPriorityRequired forAxis:UILayoutConstraintAxisHorizontal];
+    [label setContentCompressionResistancePriority:UILayoutPriorityRequired forAxis:UILayoutConstraintAxisHorizontal];
+    _label = label;
+
+    NSAssert(_labelsStackView != nil, @"labelsStackView is nil");
+    [self.labelsStackView addArrangedSubview:label];
+}
+
+- (void)layoutSubviews
+{
+    [super layoutSubviews];
+    
+    [self updateSeparatorInsets];
+}
+
+- (void)setSourceSelected:(BOOL)sourceSelected
+{
+    if (_sourceSelected != sourceSelected) {
+        _sourceSelected = sourceSelected;
+        self.accessoryType = sourceSelected ? UITableViewCellAccessoryCheckmark : UITableViewCellAccessoryNone;
+    }
+}
+
+- (void)setTitle:(NSString *)title
+{
+    if (_title != title) {
+        _title = [title copy];
+        self.label.text = title;
+    }
+}
+
+- (void)setSourceHierarchyIndentation:(NSUInteger)sourceHierarchyIndentation
+{
+    if (_sourceHierarchyIndentation != sourceHierarchyIndentation) {
+        _sourceHierarchyIndentation = sourceHierarchyIndentation;
+        self.leadingLayoutConstraintForContentViewIndentation.constant = sourceHierarchyIndentation * MenuItemSourceCellHierarchyIdentationWidth;
+        [self updateSeparatorInsets];
+    }
+}
+
+- (void)updateSeparatorInsets
+{
+    CGFloat left = self.sourceHierarchyIndentation * MenuItemSourceCellHierarchyIdentationWidth;
+    left += self.stackView.layoutMargins.left;
+    self.separatorInset = UIEdgeInsetsMake(0, left, 0, 0);
+}
+
+- (CGRect)drawingRectForLabel
+{
+    CGRect rect = [self convertRect:self.label.frame fromView:self.label.superview];
```

Comment:
Not sure why this is necessary.  The label's `superview` is the `labelsStackView` and `[self convertRect:self.label.frame fromView:self.label.superview];` returns a rect with the same value as the `labelsStackView`'s frame. It doesn't look like we're adding anything else to the stack view.  Would it make sense to just use its frame from the get go?